docs/reference/analysis/common-uses.asciidoc

```diff
@@ -0,0 +1,98 @@
+== Common uses for analysis
+++++
+<titleabbrev>Common uses</titleabbrev>
+++++
+
+Analyzing full text often means analyzing natural, human language. While each
```

Comment:
I was trying to avoid a description of "full text search" as that is sort of covered in https://www.elastic.co/guide/en/elasticsearch/reference/master/analysis.html

However, I can expand it a bit more as part of relocating this concept content to the top-level analysis page.